module/hotspot/hsperfdata/hsperfdata.go

```diff
@@ -89,59 +162,67 @@ func (m *MetricSet) buildMapStr(entries []PerfDataEntry) common.MapStr {
 	return event
 }
 
-func (m *MetricSet) publishAll() (common.MapStr, error) {
-	f, err := os.Open(m.hsPerfDataPath)
+func (p *ProcStats) publishAll() (common.MapStr, error) {
+	f, err := os.Open(p.hsPerfDataPath)
 	if err != nil {
 		return nil, err
```

Comment:
Thank you for your pull request!
But we should think the process which is owned by other user.

For example, Elasticsearch works on `elastic` user, and HSBeat works on `hsbeat` user, HSBeat cannot read the hsperfdata which is owned by Elasticsearch.

Could you check the process owner?
And we should skip this check when HSBeat works on admin user (e.g. `root` on Linux and `Administrator` on Windows).